src/generators/python/renderers/ClassRenderer.ts

```diff
@@ -81,27 +82,38 @@ export const PYTHON_DEFAULT_CLASS_PRESET: ClassPresetType<PythonOptions> = {
     const properties = model.properties || {};
     let body = '';
     if (Object.keys(properties).length > 0) {
-      const assigments = Object.values(properties).map((property) => {
+      const assignments = Object.values(properties).map((property) => {
+        let assignment: string;
+        if (property.property instanceof ConstrainedReferenceModel) {
+          assignment = `self._${property.propertyName}: ${property.property.type} = ${property.property.type}(input["${property.propertyName}"])`;
+        } else {
+          assignment = `self._${property.propertyName}: ${property.property.type} = input["${property.propertyName}"]`;
+        }
         if (!property.required) {
-          return `if hasattr(input, '${property.propertyName}'):\n\tself._${property.propertyName} = input.${property.propertyName}`;
+          return `if hasattr(input, "${property.propertyName}"):\n\t${assignment}`;
```

Comment:
Tabs are not recommended for indentation, I'd suggest to use 2 or 4 spaces instead, but consistently.

Reply:
https://github.com/asyncapi/modelina/pull/1856 clears this up :v: